sample_fast_api/apps/auth/authentication.py

```diff
@@ -0,0 +1,65 @@
+from datetime import datetime, timedelta
+from typing import Optional
+
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+from jose import JWTError, jwt
+
+from sample_fast_api.apps.users.models import User
+from sample_fast_api.apps.users.schemas import pwd_context
+from sample_fast_api.config import settings
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+
+def validate_token(token: str = Depends(oauth2_scheme)):
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, settings.SECRET_KEY, algorithms=[settings.ALGORITHM])
+        email: str = payload.get("sub")
+        if email is None:
+            raise credentials_exception
+    except JWTError:
+        raise credentials_exception
+
+
+async def authenticate_user(form_data):
+    user = await User.query.where(User.name == form_data.username).gino.first()
+    exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Incorrect username or password",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        if not user or not pwd_context.verify(form_data.password, user.password):
+            raise exception
+    except ValueError:
+        raise exception
+    return user
+
+
+class Token:
+    def __init__(self, user):
+        self.user = user
+
+    def create(self):
+        access_token_expires = timedelta(minutes=settings.ACCESS_TOKEN_EXPIRE_MINUTES)
+        access_token = self._create_access_token(
+            data={"sub": self.user.email}, expires_delta=access_token_expires
+        )
+
+        return {"access_token": access_token, "token_type": "bearer"}
+
+    def _create_access_token(self, data: dict, expires_delta: Optional[timedelta] = None):
+        to_encode = data.copy()
+        if expires_delta:
+            expire = datetime.utcnow() + expires_delta
+        else:
+            expire = datetime.utcnow() + timedelta(minutes=15)
```

Comment:
This `15` number can be a constant here at file scope or a envvar configuration

Reply:
Sorry, I forgot to include from config